homework/index.js

```diff
@@ -30,20 +30,41 @@
   }
 
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const li = createAndAppend('li', ul, { class: 'list-info' });
+    const table = createAndAppend('table', li);
+    const trName = createAndAppend('tr', table);
+    createAndAppend('th', trName, { text: 'Repository: ' });
+    const tdName = createAndAppend('td', trName);
+    createAndAppend('a', tdName, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+    const trDesc = createAndAppend('tr', table);
+    createAndAppend('th', trDesc, { text: 'Description: ' });
+    createAndAppend('td', trDesc, { text: repo.description });
+    const trFork = createAndAppend('tr', table);
+    createAndAppend('th', trFork, { text: 'Forks: ' });
+    createAndAppend('td', trFork, { text: repo.forks_count });
+    const trUpdate = createAndAppend('tr', table);
+    createAndAppend('th', trUpdate, { text: 'Last Update: ' });
+    createAndAppend('td', trUpdate, { text: repo.updated_at });
```

Comment:
I see some repetition of code here. Do you think you could create a function to make your code DRY? Of course, you would need to do something special for the first row.

Reply:
The repetition I see is:

1. Create and append a `<tr>`.
2. Create and append a `<th>` and set its text to `labelText`.
3. Create and append a `<td`> and set its text to `value`.

You do this for the 2nd, 3rd and 4th rows. The only thing that is different is the actual text for the `<th>` and the `<td>`. Whenever you see repetition like this, you have an opportunity to implement the DRY principle. You just need to identify what is different and introduce parameters to the function through which you pass on the variable parts.   

So you could write a function that takes three parameters:

1. the `table` to which to add the row
2. the label text
3. the repository value

```js
function addRow(table, labelText, value) {
  const tr = createAndAppend('tr', table);
  createAndAppend('th', tr, ...);
  createAndAppend('td', tr, ...);
  return tr;
}
```

For instance, for the second row you could call this function like this:

```js
addRow(table, 'Description: ', repo.description);
```

Even though the first row requires an `<a>` instead of simple text, you can still use that function, but you need to do some fixing up afterwards:

```js
const firstRow = addRow(table,  'Repository', '');
createAndAppend('a', firstRow.lastChild, ...);
```

The trick here is to pass an empty string for `value` which we will fix later. This works, because we return the created `tr` as the return value from `addRow()`. This trick with dealing with the first row in a special way will usually only pop up in your head when you look at your code again the next day or so. At least, that's how it usually happens for me.

Reply:
Thank you so much for your feedback, I will try to update my code accordingly.